lib/resty/redis.lua

```diff
@@ -107,7 +65,7 @@ end
 
 
 function _M.set_timeout(self, timeout)
-    local sock = self.sock
+    local sock = rawget(self, "sock")
```

Comment:
Okay, triggering `__index` is inefficient per se. So `rawget` should be the best approach anyway.